conda/models/match_spec.py

```diff
@@ -183,7 +183,7 @@ def from_dist_str(cls, dist_str):
         if '::' in dist_str:
```

Comment:
I guess after line 182 we should start adding in 

```
        if dist_str.endswith(CONDA_PACKAGE_EXTENSION_V2):
            dist_str = dist_str[:-len(CONDA_PACKAGE_EXTENSION_V2)]
```

Reply:
Although another best practice is to use string slices instead of `.endswith()`, again for performance reasons. See comment at https://github.com/conda/conda/issues/9812#issuecomment-610605261.